### PR TITLE
swarm/pss: Deduplicate messages when processing

### PR DESCRIPTION
**Pull request by nolash**
https://github.com/ethersphere/go-ethereum/issues/304

### CODE REVIEW

File: swarm/pss/pss.go [hunk 1]
```diff
@@ -294,12 +300,14 @@ func (self *Pss) handlePssMsg(msg interface{}) error {
 				return errors.New("Invalid TTL")
 			}
 			log.Trace("pss was for someone else :'( ... forwarding", "pss", common.ToHex(self.BaseAddr()))
-			return self.forward(pssmsg)
+			self.checkFwdCache([]byte{}, digest, true)
```
Comment (janos) [inline swarm/pss/pss.go]: Why is checkFwdCache called without checking the return value?

File: swarm/pss/pss.go [hunk 2]
```diff
@@ -294,12 +300,14 @@ func (self *Pss) handlePssMsg(msg interface{}) error {
...
+			return self.forward(pssmsg, digest)
 		}
 		log.Trace("pss for us, yay! ... let's process!", "pss", common.ToHex(self.BaseAddr()))
 
-		if !self.process(pssmsg) {
-			err = self.forward(pssmsg)
+		if !self.process(pssmsg, digest) {
+			self.checkFwdCache([]byte{}, digest, true)
```
Comment (janos) [inline swarm/pss/pss.go]: The same as above https://github.com/ethersphere/go-ethereum/pull/314/files#diff-2d1fd9a8750299d8ae63031e6b30be89R303.

**Review by janos — APPROVED** [review]

**Review by zelig — COMMENTED** [review]
@nolash please list the changes requested as per convo

**Comment by nolash** [comment]
Deferred to resolution of https://github.com/ethersphere/go-ethereum/issues/339

**Pull request closed without merging**